app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,65 @@
+class FoodsController < ApplicationController
+  before_action :set_food, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
+  before_action :check_owner, only: %i[edit update destroy]
+
+  # GET /foods
+  def index
+    @foods = current_user.foods
+  end
+
+  # GET /foods/1
+  def show; end
+
+  # GET /foods/new
+  def new
+    @food = current_user.foods.new
+  end
+
+  # GET /foods/1/edit
+  def edit; end
+
+  # POST /foods
+  def create
+    @food = current_user.foods.new(food_params)
+
+    if @food.save
+      redirect_to @food, notice: 'Food item was successfully created.'
+    else
+      render :new
```

Comment:
- [ ] Let's ensure to optimize our database queries by introducing solutions to **N + 1** problem in the code so as to improve the app's performance. For instance instead of fetching all books here:
```rails
class BooksController < ApplicationController
  def index
    @books = Book.all
  end
end
```

We can do:
```rails
class BooksController < ApplicationController
  def index
    @books = Book.includes(:author).all
  end
end
```
The last code will fetch books and the associated authors in a single instead making two different queries. This improves the performance.